emu_sv/custom_callback_implementations.py

```diff
@@ -63,6 +84,27 @@ def qubit_occupation_sv_den_mat_impl(
     return occupation.cpu()
 
 
+def choose_correlation_matrix_impl(
+    self: CorrelationMatrix,
+    *,
+    config: EmulationConfig,
```

Comment:
I suggest using `**kwargs` for `config` and `hamiltonian`, it feels better in that context because it is generic